certmanager/util.go

```diff
@@ -18,10 +18,40 @@ import (
 	"crypto"
 	"crypto/x509"
 	"encoding/pem"
+	"net/http"
 
 	"github.com/golang/glog"
 )
 
+const (
+	// The GCE metadata service URI to get identity token.
+	identityUri = "http://metadata/computeMetadata/v1/instance/service-accounts/default/identity"
+)
+
+// Get the GCE VM identity jwt token from its metadata server.
+// Note: this function only works in a GCE VM environment.
+func GetIdentityToken(aud string) ([]byte, error) {
```

Comment:
Could you instead have an interface as follows:

type TokenFetcher interface {
       TokenFetch(...)
}

type GcpTokenFetcher struct {
         const identityUri = "http://...."
}

func TokenFetch() ([]byte, error) {
}

This will make it very easy to incorporate a new identity to be implemented and tested.